librz/type/serialize_functions.c

```diff
@@ -60,6 +60,7 @@ static RzCallable *get_callable_type(RzTypeDB *typedb, Sdb *sdb, const char *nam
 		goto error;
 	}
 
+	char arg_name[32];
```

Comment:
i'd place this inside the `for` loop as well.

Reply:
Why? There is a need of only one static buffer so it can be allocated on stack only once (depending on the compiler of course).

Reply:
Because that's compiler's work. They are smart enough to allocate that just one time and not on every loop iteration. It's better to define things in the smallest scope needed (e.g. otherwise you could also put `char *values` at line 65 here, or other things...).

When I see `char arg_name[32]` here I don't know where this var is used exactly unless I read the whole function after this line. If I see this same definition inside the loop I know for sure that nothing else outside of the loop can use this buffer.

Reply:
OK, moved then.